server/src/main/scala/org/http4s/server/middleware/GZip.scala

```diff
@@ -67,4 +62,36 @@ object GZip {
     0.toByte,                           // Extra flags
     0.toByte)                           // Operating system
   )
+
+  private final case class TrailerGen(crc: CRC32 = new CRC32(), var inputLength: Int = 0)
```

Comment:
I don't know that I'd call this a case class with a mutable val and a var.  The mutability is justified here, but I think it's just a class.